tests/console/yast2_apparmor.pm

```diff
@@ -66,95 +58,79 @@ sub run {
     assert_screen 'yast2_apparmor_configuration_manage_existing_profiles';
     send_key 'ret';
     assert_screen 'yast2_apparmor_configuration_manage_existing_profiles_edit_add';
-    send_key 'alt-i';
-    wait_still_screen(1);
-    send_key 'alt-a';
-    wait_still_screen(1);
-    send_key 'alt-f';
-    wait_still_screen(1);
+    wait_screen_change { send_key 'alt-i' };
+    wait_screen_change { send_key 'alt-a' };
+    wait_screen_change { send_key 'alt-f' };
     assert_screen 'yast2_apparmor_configuration_manage_existing_profiles_edit_add_file';
-    send_key 'ret';
-    wait_still_screen(1);
-    send_key 'alt-e';
-    wait_still_screen(1);
+    wait_screen_change { send_key 'ret' };
+    wait_screen_change { send_key 'alt-e' };
     type_string 'I_added_this_profile';
-    wait_still_screen(1);
-    send_key 'alt-p';
-    wait_still_screen(1);
-    send_key 'down';
-    send_key 'up';
-    send_key 'spc';
-    send_key 'down';
-    send_key 'spc';
+    wait_screen_change { send_key 'alt-p' };
+    wait_screen_change { send_key 'down' };
+    wait_screen_change { send_key 'up' };
+    wait_screen_change { send_key 'spc' };
+    wait_screen_change { send_key 'down' };
+    wait_screen_change { send_key 'spc' };
 
     # confirm with cancel
-    send_key 'alt-c';
-    wait_still_screen(1);
+    wait_screen_change { send_key 'alt-c' };
 
     # now add a directory with permission for read and write
-    send_key 'alt-a';
-    wait_still_screen(1);
-    send_key 'alt-d';
-    wait_still_screen(1);
-    send_key 'alt-b';
-    wait_still_screen(1);
-    send_key 'alt-c';
-    wait_still_screen(1);
-    send_key 'alt-e';
-    wait_still_screen(1);
+    wait_screen_change { send_key 'alt-a' };
+    wait_screen_change { send_key 'alt-d' };
+    wait_screen_change { send_key 'alt-b' };
+    wait_screen_change { send_key 'alt-c' };
+    wait_screen_change { send_key 'alt-e' };
     type_string '/tmp';
-    wait_still_screen(1);
-    send_key 'alt-p';
-    wait_still_screen(1);
-    send_key 'down';
-    send_key 'up';
-    send_key 'spc';
-    send_key 'down';
-    send_key 'spc';
-    send_key 'alt-o';
-    wait_still_screen(1);
+    wait_screen_change { send_key 'alt-p' };
+    wait_screen_change { send_key 'down' };
+    wait_screen_change { send_key 'up' };
+    wait_screen_change { send_key 'spc' };
+    wait_screen_change { send_key 'down' };
+    wait_screen_change { send_key 'spc' };
+    wait_screen_change { send_key 'alt-o' };
```

Comment:
this is a huge list of actions with no assert_screen in between means that if there is something going wrong it would be very hard to find the key press that went wrong. I suggest to make some intermediate steps assert_screen instead.

@rwx788 ?